backend/open_webui/models/knowledge.py

```
@@ -89,127 +114,615 @@ class KnowledgeResponse(KnowledgeModel):
+
+                return self.get_knowledge_by_id(id=id, db=db)
+        except Exception as e:
+            log.exception(e)
+            return None
+
     def update_knowledge_by_id(
-        self, id: str, form_data: KnowledgeForm, overwrite: bool = False
+        self,
+        id: str,
+        form_data: KnowledgeForm,
+        overwrite: bool = False,
```

Ruff (0.15.2)

[warning] Unused method argument: `overwrite`

(ARG002)